### ISSUE TITLE
2020 : SIGIR: Joint Item Recommendation and Attribute Inference: An Adaptive Graph Convolutional Network Approach

### ISSUE THREAD

**Issue by karan96** [fani-lab/OpeNTF#170]
Title: - SIGIR: Joint Item Recommendation and Attribute Inference: An Adaptive Graph Convolutional Network Approach
Year: - 2020
Venue: - 43rd International ACM SIGIR conference on research and development in Information Retrieval

Main Problem
The authors argues that recommendation methods such as Collabortive Filtering and GCNs do not perform very well when it comes to values missing some of the attribute information. In a real world setting this is frequently observed when a user's attributes are incomplete. The authors target the joint item recommendation problem and attribute inference problem under GCN based approach.
Item Recommendation Problem: - Recommending an item to the user.
Attribute Inference Problem: - Missing attributes for a particular user.

Related Work
Author states multiple problems with the existing work that serves a basis for this research work: - 
- As CF methods suffer from Cold Start Problem, most CF based methods assumes that the attributed information is complete. As conventional methods could not tackle the missing attribute problem, they usually employ a preprocessing step to tackle that.
- GCN Methods typically consdier the above two problems separately and authors' work differs from these as they simultaneously consider the propagation of collabortive signals and node attributes with missing attributes.
- The attribute inference problem could also be labelled as a supervised classficiation problem by taking remaining attributes as input features and attribute values for current attributes as labels but this approach fails when attribute information is incomplete with large number of missing values.
- Some recent approaches that do consider these two problems together suffer from high time complexity and could not encode the complex global structure very well.

Proposed Approach
The author proposes a model named AGCN which contains two learning modules: - graph learning module and attribute update module. At each iteration, graph learning module takes the predicted attribute value and learns network parameters which is then fed to the attribute update module  that is then designed to infer missing attribute values which then updates these and are sent back to graph learning module until the model converges. 

Input
Users, Items and their respective attribute values. 

Output
Recommended item for a particular user.

Dataset
Amazon-Video Games
Movilens -1M
Movielens -20M

Code
https://github.com/yimutianyang/AGCN

Pros and Cons

The paper introduces a new approach to the missing attribute problem
Provides good comparison against baselines. 

Currently in process of implementing this on our toy dataset.

**Comment by hosseinfani** [comment]
@soroush-ziaeinejad

**Comment by karan96** [comment]
@hosseinfani Greetings Dr. Fani,
 In order to implement this work on our toy dataset, I am trying to understand the files they are using as the dataset in their work. They have not explained how they are preprocessing the data so I am trying to understand myself. Here is a summary of the files they are using from what I have understood: - 

traindata	#	Training Data
valdata		#	Validation Data
testdata	#	Test Data
user_items	#	User - Items
A_indexs	#	User/Items(Uncertain) to indexes
A_values	#	Couldn't understand how they are creating this.
item_attributes	#     Attributes of skills and location - Needs discussion
item_attributes_missing	#	Not Required and Couldn't understand how they are creating this.
existing_item_list	#	Not Required
userset	#	User ids
itemset	#	Item ids(Skills, Location)
missing_item_list	#	Not Required

I have confusion on how should the data be prepared and want to have a discussion with you.

**Comment by karan96** [comment]
An Update on the above data files based on the discussion we had today:  -
users: - experts / location
user's attirbutes: - skills
items: - location / users

traindata # Training Data
valdata # Validation Data
testdata # Test Data
user_items # User - Items -- User to location set where ({userid:{location_ids}})
A_indexs # User/Items(Uncertain) to indexes - Adjacency Matrix of user, items in the form: - [[user_id_0, item_id_0], [item_id_0, user_id_0], ........]
A_values # Couldn't understand how they are creating this. - Adjacency Values - Not sure how they are creating this
item_attributes # Attributes of skills and location - Needs discussion - Not focusing/including in our work
item_attributes_missing # Not Required and Couldn't understand how they are creating this. - Not Required
existing_item_list # Not Required
userset # User ids - Expert ids
itemset # Item ids - Location ids
missing_item_list # Not Required

**Comment by karan96** [comment]
@hosseinfani  Dr. Fani, 
I require your presence in the lab tomorrow i.e. Saturday 17 Sept. Kindly let me know when you are available and I will come to the lab.
